VolumeFailedDelete  PV remains after PVC delete

**What happened**:
We deployed the azurefiles-csi driver to a TKG K8s cluster using helm, and deploy a sts from the examples, then when delete the sts and pvc, the pv remains behind in a released mode, 

**What you expected to happen**:
We expect the PV to be removed after the PV deletion.

**How to reproduce it**:
After deploying the azurefile-csi via helm follow option 2 from this page
https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/deploy/example/e2e_usage.md

**Anything else we need to know?**:
the csi-azurefile-controller pods go into a CrashLoopBackOff 
```
 Warning  BackOff    3m2s (x34 over 10m)  kubelet            Back-off restarting failed container azurefile in pod csi-azurefile-controller-5b4d749497-cxgmf_kube-system(3a4b2326-2e38-4141-9de5-eeaf6d657146)
**Environment**:
```
The PV logs give this warning
```
Warning  VolumeFailedDelete  12m    file.csi.azure.com_tkg-01-kp5jm-5598f_fdf642c8-b608-492b-80f3-83c3ffcd14e0  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  12m    file.csi.azure.com_tkg-01-kp5jm-5598f_78999074-6e18-4eaf-a480-337cdbf342c2  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  11m    file.csi.azure.com_tkg-01-kp5jm-5598f_746457c4-b380-4cda-812f-d1c446bc4d01  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  11m    file.csi.azure.com_tkg-01-kp5jm-5598f_2a3e54af-9952-4c03-ba49-ae702c225ec7  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  9m52s  file.csi.azure.com_tkg-01-kp5jm-5598f_9e7ada71-1bfe-421b-8d13-c412b2b850ff  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  7m59s  file.csi.azure.com_tkg-01-kp5jm-5598f_252dd830-247c-4cf1-a671-a63d62acb8bc  rpc error: code = Unavailable desc = error reading from server: EOF
  Warning  VolumeFailedDelete  4m52s  file.csi.azure.com_tkg-01-kp5jm-5598f_6701c709-6d6e-47a6-9489-44a465e9cf06  rpc error: code = Unavailable desc = error reading from server: EOF
```

If we delete the pv manually the pv get stuck terminating, when have to delete the finilizer for it to clear.
At which point the controller pods go into a running state again.

<!-- 
*Run following command to get CSI driver version:
kubectl get po -n kube-system -o yaml | grep mcr | grep azurefile
-->
- CSI Driver version: v1.29.2
- Kubernetes version (use `kubectl version`): "v1.27.5
- OS (e.g. from /etc/os-release): PhotonOS
- Kernel (e.g. `uname -a`): 
- Install tools:
- Others:

Here's the output,
We inadvertently deleted the pod, so waited till the new pod was created, same errors.

```
W0217 05:41:40.422287       1 main.go:83] nodeid is empty
I0217 05:41:40.422571       1 main.go:140] set up prometheus server on [::]:29614
I0217 05:41:40.423373       1 azurefile.go:361]
DRIVER INFORMATION:
-------------------
Build Date: "2023-11-23T13:24:54Z"
Compiler: gc
Driver Name: file.csi.azure.com
Driver Version: v1.29.2
Git Commit: 4f98c74f750742c8541b7cb266b4f48bdac1d430
Go Version: go1.21.3
Platform: linux/amd64

Streaming logs below:
I0217 05:41:40.423396       1 azurefile.go:364] driver userAgent: file.csi.azure.com/v1.29.2 OSS-helm
I0217 05:41:40.423551       1 azure.go:69] set QPS(25.000000) and QPS Burst(50) for driver kubeClient
I0217 05:41:40.423974       1 azure.go:84] reading cloud config from secret kube-system/azure-cloud-provider
I0217 05:41:40.432191       1 azure.go:91] InitializeCloudFromSecret: failed to get cloud config from secret kube-system/azure-cloud-provider: failed to get secret kube-system/azure-cloud-provider: secrets "azure-cloud-provider" not found
I0217 05:41:40.432215       1 azure.go:96] could not read cloud config from secret kube-system/azure-cloud-provider
I0217 05:41:40.432224       1 azure.go:106] use default AZURE_CREDENTIAL_FILE env var: /etc/kubernetes/azure.json
W0217 05:41:40.432245       1 azure.go:111] load azure config from file(/etc/kubernetes/azure.json) failed with open /etc/kubernetes/azure.json: no such file or directory
I0217 05:41:40.432252       1 azure.go:123] no cloud config provided, error: failed to get secret kube-system/azure-cloud-provider: secrets "azure-cloud-provider" not found, driver will run without cloud config
I0217 05:41:40.432262       1 azure.go:147] starting controller server...
I0217 05:41:40.432269       1 azurefile.go:369] cloud: , location: , rg: , VnetName: , VnetResourceGroup: , SubnetName:
I0217 05:41:40.434666       1 mount_linux.go:285] 'umount /tmp/kubelet-detect-safe-umount2286003081' failed with: exit status 32, output: umount: /tmp/kubelet-detect-safe-umount2286003081: must be superuser to unmount.
I0217 05:41:40.434695       1 mount_linux.go:287] Detected umount with unsafe 'not mounted' behavior
I0217 05:41:40.434762       1 driver.go:93] Enabling controller service capability: CREATE_DELETE_VOLUME
I0217 05:41:40.434775       1 driver.go:93] Enabling controller service capability: CREATE_DELETE_SNAPSHOT
I0217 05:41:40.434780       1 driver.go:93] Enabling controller service capability: EXPAND_VOLUME
I0217 05:41:40.434784       1 driver.go:93] Enabling controller service capability: SINGLE_NODE_MULTI_WRITER
I0217 05:41:40.434788       1 driver.go:93] Enabling controller service capability: CLONE_VOLUME
I0217 05:41:40.434793       1 driver.go:112] Enabling volume access mode: SINGLE_NODE_WRITER
I0217 05:41:40.434797       1 driver.go:112] Enabling volume access mode: SINGLE_NODE_READER_ONLY
I0217 05:41:40.434801       1 driver.go:112] Enabling volume access mode: SINGLE_NODE_SINGLE_WRITER
I0217 05:41:40.434804       1 driver.go:112] Enabling volume access mode: SINGLE_NODE_MULTI_WRITER
I0217 05:41:40.434808       1 driver.go:112] Enabling volume access mode: MULTI_NODE_READER_ONLY
I0217 05:41:40.434811       1 driver.go:112] Enabling volume access mode: MULTI_NODE_SINGLE_WRITER
I0217 05:41:40.434815       1 driver.go:112] Enabling volume access mode: MULTI_NODE_MULTI_WRITER
I0217 05:41:40.434819       1 driver.go:103] Enabling node service capability: STAGE_UNSTAGE_VOLUME
I0217 05:41:40.434824       1 driver.go:103] Enabling node service capability: SINGLE_NODE_MULTI_WRITER
I0217 05:41:40.434828       1 driver.go:103] Enabling node service capability: VOLUME_MOUNT_GROUP
I0217 05:41:40.434831       1 driver.go:103] Enabling node service capability: GET_VOLUME_STATS
I0217 05:41:40.434959       1 server.go:118] Listening for connections on address: &net.UnixAddr{Name:"//csi/csi.sock", Net:"unix"}
I0217 05:41:41.196779       1 utils.go:76] GRPC call: /csi.v1.Identity/GetPluginInfo
I0217 05:41:41.196814       1 utils.go:77] GRPC request: {}
I0217 05:41:41.198429       1 utils.go:83] GRPC response: {"name":"file.csi.azure.com","vendor_version":"v1.29.2"}
I0217 05:41:41.203173       1 utils.go:76] GRPC call: /csi.v1.Identity/GetPluginCapabilities
I0217 05:41:41.203201       1 utils.go:77] GRPC request: {}
I0217 05:41:41.203328       1 utils.go:83] GRPC response: {"capabilities":[{"Type":{"Service":{"type":1}}}]}
I0217 05:41:41.205385       1 utils.go:76] GRPC call: /csi.v1.Controller/ControllerGetCapabilities
I0217 05:41:41.205400       1 utils.go:77] GRPC request: {}
I0217 05:41:41.205472       1 utils.go:83] GRPC response: {"capabilities":[{"Type":{"Rpc":{"type":1}}},{"Type":{"Rpc":{"type":5}}},{"Type":{"Rpc":{"type":9}}},{"Type":{"Rpc":{"type":13}}},{"Type":{"Rpc":{"type":7}}}]}
I0217 05:41:58.873603       1 utils.go:76] GRPC call: /csi.v1.Controller/DeleteVolume
I0217 05:41:58.873692       1 utils.go:77] GRPC request: {"secrets":"***stripped***","volume_id":"##pvc-5ad5368b-799c-4f9b-a512-1da13db7b6d1###default"}
I0217 05:41:58.873938       1 azurefile.go:597] got storage account(iedk8sazfs) from secret
W0217 05:41:58.992511       1 azurefile.go:912] DeleteFileShare(pvc-5ad5368b-799c-4f9b-a512-1da13db7b6d1) on account() failed with error(storage: service returned error: StatusCode=404, ErrorCode=ShareNotFound, ErrorMessage=The specified share does not exist.
RequestId:0627a69b-601a-0056-0964-61b4c5000000
Time:2024-02-17T05:41:58.9807592Z, RequestInitiated=Sat, 17 Feb 2024 05:41:58 GMT, RequestId=0627a69b-601a-0056-0964-61b4c5000000, API Version=2018-03-28, QueryParameterName=, QueryParameterValue=), return as success
I0217 05:41:58.992573       1 controllerserver.go:719] azure file(pvc-5ad5368b-799c-4f9b-a512-1da13db7b6d1) under subsID() rg() account() volume(##pvc-5ad5368b-799c-4f9b-a512-1da13db7b6d1###default) is deleted successfully
I0217 05:41:58.992595       1 azurefile.go:1039] remove tag(skip-matching) on account() subsID(), resourceGroup()
I0217 05:41:58.992723       1 azure_metrics.go:115] "Observed Request Latency" latency_seconds=0.118714537 request="azurefile_csi_driver_controller_delete_volume" resource_group="" subscription_id="" source="file.csi.azure.com" volumeid="##pvc-5ad5368b-799c-4f9b-a512-1da13db7b6d1###default" result_code="failed_csi_driver_controller_delete_volume"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x173044a]

goroutine 65 [running]:
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*Cloud).RemoveStorageAccountTag(0xc0000f5800, {0x21c5e90, 0xc0004ff170}, {0x0, 0x0}, {0x0, 0x0}, {0xc00024e041, 0x0}, {0x1ed4bac, ...})
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_storageaccount.go:661 +0x8a
sigs.k8s.io/azurefile-csi-driver/pkg/azurefile.(*Driver).RemoveStorageAccountTag(0xc000574000, {0x21c5e90, 0xc0004ff170}, {0x0, 0x0}, {0x0, 0x0}, {0xc00024e041, 0x0}, {0x1ed4bac, ...})
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/pkg/azurefile/azurefile.go:1041 +0x4dd
sigs.k8s.io/azurefile-csi-driver/pkg/azurefile.(*Driver).DeleteVolume(0xc000574000, {0x21c5e90, 0xc0004ff170}, 0xc000122ac0)
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/pkg/azurefile/controllerserver.go:720 +0xc10
github.com/container-storage-interface/spec/lib/go/csi._Controller_DeleteVolume_Handler.func1({0x21c5e90, 0xc0004ff170}, {0x1d91820?, 0xc000122ac0})
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:6438 +0x72
sigs.k8s.io/azurefile-csi-driver/pkg/csi-common.logGRPC({0x21c5e90, 0xc0004ff170}, {0x1d91820?, 0xc000122ac0?}, 0xc00023e340, 0xc0000b6108)
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/pkg/csi-common/utils.go:79 +0x3a9
github.com/container-storage-interface/spec/lib/go/csi._Controller_DeleteVolume_Handler({0x1e85580?, 0xc000574000}, {0x21c5e90, 0xc0004ff170}, 0xc000568800, 0x1fe95f0)
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:6440 +0x135
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005741e0, {0x21c5e90, 0xc0004fe4e0}, {0x21ce6a0, 0xc0004a5380}, 0xc0000bc360, 0xc0000bbf50, 0x30fd818, 0x0)
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/google.golang.org/grpc/server.go:1343 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0005741e0, {0x21ce6a0, 0xc0004a5380}, 0xc0000bc360)
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/google.golang.org/grpc/server.go:1737 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/google.golang.org/grpc/server.go:986 +0x86
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 47
        /mnt/vss/_work/1/go/src/sigs.k8s.io/azurefile-csi-driver/vendor/google.golang.org/grpc/server.go:997 +0x145
```

--- a/pkg/azurefile/azurefile.go
+++ b/pkg/azurefile/azurefile.go
@@ -438,7 +438,7 @@ func (d *Driver) getFileShareQuota(ctx context.Context, subsID, resourceGroupNam
 		if err != nil {
 			return -1, err
 		}
-		fileClient, err := newAzureFileClient(&d.cloud.Environment, &retry.Backoff{Steps: 1}, accountName, accountKey)
+		fileClient, err := newAzureFileClient(accountName, accountKey, d.getStorageEndPointSuffix(), &retry.Backoff{Steps: 1})
 		if err != nil {
 			return -1, err
 		}
@@ -908,7 +908,7 @@ func (d *Driver) CreateFileShare(ctx context.Context, accountOptions *azure.Acco
 			if rerr != nil {
 				return true, rerr
 			}
-			fileClient, rerr := newAzureFileClient(&d.cloud.Environment, &retry.Backoff{Steps: 1}, accountName, accountKey)
+			fileClient, rerr := newAzureFileClient(accountName, accountKey, d.getStorageEndPointSuffix(), &retry.Backoff{Steps: 1})
 			if rerr != nil {
 				return true, rerr
 			}
@@ -934,7 +934,7 @@ func (d *Driver) DeleteFileShare(ctx context.Context, subsID, resourceGroup, acc
 			if rerr != nil {
 				return true, rerr
 			}
-			fileClient, rerr := newAzureFileClient(&d.cloud.Environment, &retry.Backoff{Steps: 1}, accountName, accountKey)
+			fileClient, rerr := newAzureFileClient(accountName, accountKey, d.getStorageEndPointSuffix(), &retry.Backoff{Steps: 1})
 			if rerr != nil {
 				return true, rerr
 			}
@@ -975,7 +975,7 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 			if rerr != nil {
 				return true, rerr
 			}
-			fileClient, rerr := newAzureFileClient(&d.cloud.Environment, &retry.Backoff{Steps: 1}, accountName, accountKey)
+			fileClient, rerr := newAzureFileClient(accountName, accountKey, d.getStorageEndPointSuffix(), &retry.Backoff{Steps: 1})
 			if rerr != nil {
 				return true, rerr
 			}
@@ -1078,6 +1078,9 @@ func (d *Driver) GetTotalAccountQuota(ctx context.Context, subsID, resourceGroup
 
 // RemoveStorageAccountTag remove tag from storage account
 func (d *Driver) RemoveStorageAccountTag(ctx context.Context, subsID, resourceGroup, account, key string) error {
+	if d.cloud == nil {
+		return fmt.Errorf("cloud is nil")
+	}
 	// search in cache first
 	cache, err := d.skipMatchingTagCache.Get(account, azcache.CacheReadTypeDefault)
 	if err != nil {
@@ -1219,4 +1222,11 @@ func (d *Driver) SetAzureCredentials(ctx context.Context, accountName, accountKe
 		return "", fmt.Errorf("couldn't create secret %v", err)
 	}
 	return secretName, err
+}
+
+func (d *Driver) getStorageEndPointSuffix() string {
+	if d.cloud == nil || d.cloud.Environment.StorageEndpointSuffix == "" {
+		return defaultStorageEndPointSuffix
+	}
+	return d.cloud.Environment.StorageEndpointSuffix
 }

--- a/pkg/azurefile/azurefile_dataplane_client.go
+++ b/pkg/azurefile/azurefile_dataplane_client.go
@@ -23,7 +23,6 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	azs "github.com/Azure/azure-sdk-for-go/storage"
-	"github.com/Azure/go-autorest/autorest/azure"
 	"k8s.io/klog/v2"
 
 	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/fileclient"
@@ -51,9 +50,8 @@ type azureFileDataplaneClient struct {
 	*azs.FileServiceClient
 }
 
-func newAzureFileClient(env *azure.Environment, backoff *retry.Backoff, accountName, accountKey string) (azureFileClient, error) {
-	storageEndpointSuffix := env.StorageEndpointSuffix
-	if storageEndpointSuffix != "" {
+func newAzureFileClient(accountName, accountKey, storageEndpointSuffix string, backoff *retry.Backoff) (azureFileClient, error) {
+	if storageEndpointSuffix == "" {
 		storageEndpointSuffix = defaultStorageEndPointSuffix
 	}
 

--- a/pkg/azurefile/azurefile_dataplane_client_test.go
+++ b/pkg/azurefile/azurefile_dataplane_client_test.go
@@ -23,8 +23,6 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/Azure/go-autorest/autorest/azure"
-
 	"sigs.k8s.io/cloud-provider-azure/pkg/azureclients/fileclient"
 )
 
@@ -41,9 +39,7 @@ func TestCreateFileShare(t *testing.T) {
 					Name:       "devstoreaccount1",
 					RequestGiB: 10,
 				}
-				f, err := newAzureFileClient(&azure.Environment{
-					StorageEndpointSuffix: "ut",
-				}, nil, "test", "dW5pdHRlc3Q=")
+				f, err := newAzureFileClient("test", "dW5pdHRlc3Q=", "ut", nil)
 				if err != nil {
 					t.Errorf("error creating azure client: %v", err)
 				}
@@ -61,9 +57,7 @@ func TestCreateFileShare(t *testing.T) {
 }
 
 func TestNewAzureFileClient(t *testing.T) {
-	_, actualErr := newAzureFileClient(&azure.Environment{
-		StorageEndpointSuffix: "ut",
-	}, nil, "ut", "ut")
+	_, actualErr := newAzureFileClient("ut", "ut", "ut", nil)
 	if actualErr != nil {
 		expectedErr := fmt.Errorf("error creating azure client: azure: account name is not valid: it must be between 3 and 24 characters, and only may contain numbers and lowercase letters: ut")
 		if !reflect.DeepEqual(actualErr, expectedErr) {

--- a/pkg/azurefile/azurefile_test.go
+++ b/pkg/azurefile/azurefile_test.go
@@ -1414,3 +1414,56 @@ func TestGetTotalAccountQuota(t *testing.T) {
 		assert.Equal(t, test.expectedShareNum, fileShareNum, test.name)
 	}
 }
+
+func TestGetStorageEndPointSuffix(t *testing.T) {
+	d := NewFakeDriver()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	tests := []struct {
+		name           string
+		cloud          *azure.Cloud
+		expectedSuffix string
+	}{
+		{
+			name:           "nil cloud",
+			cloud:          nil,
+			expectedSuffix: "core.windows.net",
+		},
+		{
+			name:           "empty cloud",
+			cloud:          &azure.Cloud{},
+			expectedSuffix: "core.windows.net",
+		},
+		{
+			name: "cloud with storage endpoint suffix",
+			cloud: &azure.Cloud{
+				Environment: azure2.Environment{
+					StorageEndpointSuffix: "suffix",
+				},
+			},
+			expectedSuffix: "suffix",
+		},
+		{
+			name: "public cloud",
+			cloud: &azure.Cloud{
+				Environment: azure2.PublicCloud,
+			},
+			expectedSuffix: "core.windows.net",
+		},
+		{
+			name: "china cloud",
+			cloud: &azure.Cloud{
+				Environment: azure2.ChinaCloud,
+			},
+			expectedSuffix: "core.chinacloudapi.cn",
+		},
+	}
+
+	for _, test := range tests {
+		d.cloud = test.cloud
+		suffix := d.getStorageEndPointSuffix()
+		assert.Equal(t, test.expectedSuffix, suffix, test.name)
+	}
+}

--- a/pkg/azurefile/controllerserver.go
+++ b/pkg/azurefile/controllerserver.go
@@ -416,11 +416,7 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 	}
 
 	if strings.TrimSpace(storageEndpointSuffix) == "" {
-		if d.cloud.Environment.StorageEndpointSuffix != "" {
-			storageEndpointSuffix = d.cloud.Environment.StorageEndpointSuffix
-		} else {
-			storageEndpointSuffix = defaultStorageEndPointSuffix
-		}
+		storageEndpointSuffix = d.getStorageEndPointSuffix()
 	}
 
 	accountOptions := &azure.AccountOptions{
@@ -599,7 +595,7 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		diskSizeBytes := volumehelper.GiBToBytes(requestGiB)
 		klog.V(2).Infof("begin to create vhd file(%s) size(%d) on share(%s) on account(%s) type(%s) rg(%s) location(%s)",
 			diskName, diskSizeBytes, validFileShareName, account, sku, resourceGroup, location)
-		if err := createDisk(ctx, accountName, accountKey, d.cloud.Environment.StorageEndpointSuffix, validFileShareName, diskName, diskSizeBytes); err != nil {
+		if err := createDisk(ctx, accountName, accountKey, d.getStorageEndPointSuffix(), validFileShareName, diskName, diskSizeBytes); err != nil {
 			return nil, status.Errorf(codes.Internal, "failed to create VHD disk: %v", err)
 		}
 		klog.V(2).Infof("create vhd file(%s) size(%d) on share(%s) on account(%s) type(%s) rg(%s) location(%s) successfully",
@@ -1082,7 +1078,7 @@ func (d *Driver) getServiceURL(ctx context.Context, sourceVolumeID string, secre
 		return azfile.ServiceURL{}, "", err
 	}
 
-	u, err := url.Parse(fmt.Sprintf(serviceURLTemplate, accountName, d.cloud.Environment.StorageEndpointSuffix))
+	u, err := url.Parse(fmt.Sprintf(serviceURLTemplate, accountName, d.getStorageEndPointSuffix()))
 	if err != nil {
 		klog.Errorf("parse serviceURLTemplate error: %v", err)
 		return azfile.ServiceURL{}, "", err
@@ -1092,7 +1088,6 @@ func (d *Driver) getServiceURL(ctx context.Context, sourceVolumeID string, secre
 	}
 
 	serviceURL := azfile.NewServiceURL(*u, azfile.NewPipeline(credential, azfile.PipelineOptions{}))
-
 	return serviceURL, fileShareName, nil
 }
 

--- a/pkg/azurefile/nodeserver.go
+++ b/pkg/azurefile/nodeserver.go
@@ -266,11 +266,7 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 	defer d.volumeLocks.Release(lockKey)
 
 	if strings.TrimSpace(storageEndpointSuffix) == "" {
-		if d.cloud.Environment.StorageEndpointSuffix != "" {
-			storageEndpointSuffix = d.cloud.Environment.StorageEndpointSuffix
-		} else {
-			storageEndpointSuffix = defaultStorageEndPointSuffix
-		}
+		storageEndpointSuffix = d.getStorageEndPointSuffix()
 	}
 
 	// replace pv/pvc name namespace metadata in fileShareName

